include/af/defines.h

```diff
@@ -446,6 +446,13 @@ typedef enum {
 #endif
 ////////////////////////////////////////////////////////////////////////////////
 
+#if AF_API_VERSION >= 35
+typedef enum {
+    AF_MANUAL_THRESHOLD    = 0, ///< User has to define canny thresholds manually
+    AF_AUTO_OTSU_THRESHOLD = 1, ///< Determine canny algorithm thresholds using Otsu algorithm
+} af_canny_threshold;
```

Comment:
Based on what I have found in the literature, some frameworks/libraries does provide a version(not necessarily another function, could be overloaded API) of the function where threshold values are chosen automatically. Hence, the current API provides that flexibility of choosing high threshold ratio either to be computed by Otsu's algorithm or set manually.

Providing this choice via an enum instead of bool basically gives us the opportunity to add additional auto thresholding options in the future.